source/common/upstream/eds.cc

```diff
@@ -17,13 +17,14 @@
 namespace Envoy {
 namespace Upstream {
 
-EdsClusterImpl::EdsClusterImpl(const envoy::api::v2::Cluster& cluster, Runtime::Loader& runtime,
-                               Stats::Store& stats, Ssl::ContextManager& ssl_context_manager,
-                               const LocalInfo::LocalInfo& local_info, ClusterManager& cm,
-                               Event::Dispatcher& dispatcher, Runtime::RandomGenerator& random,
-                               bool added_via_api)
-    : BaseDynamicClusterImpl(cluster, cm.bindConfig(), runtime, stats, ssl_context_manager,
-                             cm.clusterManagerFactory().secretManager(), added_via_api),
+EdsClusterImpl::EdsClusterImpl(
+    const envoy::api::v2::Cluster& cluster, Runtime::Loader& runtime, Stats::Store& stats,
+    Ssl::ContextManager& ssl_context_manager, const LocalInfo::LocalInfo& local_info,
+    ClusterManager& cm, Event::Dispatcher& dispatcher, Runtime::RandomGenerator& random,
+    bool added_via_api,
+    Secret::DynamicTlsCertificateSecretProviderFactoryContextPtr secret_provider_context)
+    : BaseDynamicClusterImpl(cluster, runtime, stats, ssl_context_manager, cm, added_via_api,
+                             std::move(secret_provider_context)),
```

Comment:
I think secret_provider_context can pass as reference, not move in as unique_ptr